File: 2306-naming-a-company/2306-naming-a-company.cpp
class Solution {
public:
    long long distinctNames(vector<string>& ideas) {
        long long count = 0;
        vector<unordered_set<string>> arr(26);
        for (auto s : ideas) 
            arr[s[0] - 'a'].insert(s.substr(1));
        unordered_set<string> st;
        for (int i = 0; i < 25; i++) {
            for (int j = i + 1; j < 26; j++) {
                st.clear();
                st.insert(arr[i].begin(), arr[i].end());
                st.insert(arr[j].begin(), arr[j].end());
                count += (arr[i].size() - st.size()) * (arr[j].size() - st.size());
            }
        }
        return count * 2;
    }
};

File: 2306-naming-a-company/2306-naming-a-company.py
class Solution:
    def distinctNames(self, ideas: List[str]) -> int:
        arr = [set() for _ in range(26)]
        for idea in ideas:
            arr[ord(idea[0]) - ord('a')].add(idea[1:])
        
        count = 0
        for i in range(25):
            for j in range(i + 1, 26):
                temp = len(arr[i] & arr[j]) 
                count += 2 * (len(arr[i]) - temp) * (len(arr[j]) - temp)
                
        return count